File: 1431. Kids With the Greatest Number of Candies/kidsWithCandies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        if not candies:
            return []
        maxNum = candies[0]
        res = []
        for n in candies:
            maxNum = max(maxNum, n)
        for n in candies:
            if n + extraCandies >= maxNum:
                res.append(True)
            else:
                res.append(False)
        return res

File: 1431. Kids With the Greatest Number of Candies/kidsWithCandies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> res = new ArrayList<>();
        if (candies.length == 0) {
            return res;
        }
        
        int max = candies[0];
        for (int n: candies) {
            max = Math.max(max, n);
        }

        for (int n: candies) {
            if (n + extraCandies >= max) {
                res.add(true);
            } else {
                res.add(false);
            }
        }
        return res;
    }
}